mlx/c/stream.cpp

```diff
@@ -8,6 +8,13 @@
 #include "mlx/c/private/string.h"
 #include "mlx/c/stream.h"
 
+namespace {
+mlx_device_type to_c_device_type(mlx::core::Device::DeviceType type) {
+  static mlx_device_type map[] = {MLX_CPU, MLX_GPU};
+  return map[(int)type];
+}
+} // namespace
```

Comment:
Maybe this should be refactored into a common implementation file.. was a bit split on it, since it's like two lines of code copied from device.cpp